app/models/gallery.rb

```diff
@@ -1,7 +1,8 @@
 class Gallery < ActiveRecord::Base
   belongs_to :user
-  belongs_to :cover_icon, :class_name => Icon
+  belongs_to :cover_icon, class_name: Icon
   has_and_belongs_to_many :icons, after_add: :set_has_gallery, after_remove: :unset_has_gallery
+  has_and_belongs_to_many :ordered_icons, after_add: :set_has_gallery, after_remove: :unset_has_gallery, order: 'LOWER(keyword)', class_name: Icon
```

Comment:
I'm not sure how to do a relation scope. If you mean a scope on the icons, that means it can't be `includes`-ed, or – it can, but it'll end up being reloaded when the scope is used, I think?